cowbird/handlers/impl/magpie.py

```diff
@@ -87,7 +97,21 @@ def get_resource_id(self, resource_full_name):
         # type (str) -> str
         raise NotImplementedError
 
-    def get_resources_tree(self, resource_id):
+    def get_services_by_type(self, service_type):
+        # type: (str) -> List
```

Comment:
I put only `List` because it can get complicated considering the structure of the query response from Magpie, which can have recursive elements too sometimes.

I would get things like `List[Dict[str, Union[str, int, None, List[str], List[Dict[str,str]]]]]` or `Dict[str, Union[str, None, bool, int, List[str], List[Dict[str, str], Dict[str, Dict[str, str]]]]]`.

Considering this, is it okay to just leave the short undetailed form, or maybe I should check to create custom typings to get a shorter form?

Reply:
You can use the type `cowbird.typedefs.JSON`. Since a list is returned here, `List[JSON]` would be sufficient. At least, it conveys that its "some JSON", not just a string or a specific class.